### DISCUSSION TITLE
Is ordering on relations guaranteed?

### DISCUSSION THREAD

**Discussion by gBasil** [pocketbase/pocketbase#2800] [Q&A] [answered]
If I have a one-to-many relation, say from collection `catgirls` to `laptops`, is it guaranteed that `laptops` will be consistently ordered by the time they were added in the relation field?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the question. Could you provide a sample request and/or more details about your collections configuration?

**Reply by gBasil** [reply]
I'm looking for a way to have ordered relations. Say I have this relation:
![image](https://github.com/pocketbase/pocketbase/assets/57513430/de10628a-6095-405a-ac47-f62e416b5e6c)
If I add a relation (for example, through this UI or through code):
![image](https://github.com/pocketbase/pocketbase/assets/57513430/e67e10f7-72fe-4d8a-b09e-51c2bd04e309)
I'm wondering if they will stay in the order that they were added in, and if that is behavior I can rely on. For example, if I add the `eventCategories` `Example A`, `Example B`, and `Chased by a horde of goats` to some event in that order, can I expect that when I fetch the data for that event that they will appear in the same order? ([`Example A`, `Example B`, `Chased by a horde of goats`])

**Reply by ganigeorgiev** [reply, marked as answer]
The multiple `relation` field values are stored and returned in the same order as they were added (aka. when you fetch the record you'll get an array with the relation ids in the exact same order). You also have a drag&drop option to rearrange the selection in the Admin UI.

When performing [`expand`](https://pocketbase.io/docs/expanding-relations/), the multiple relation result in the expand is not guaranteed to be ordered since we currently don't apply an explicit `ORDER BY` clause and just eager load all records with an `IN` statement for the related ids.
